### PR TITLE
RavenDB-19399 added HttpPooledConnectionLifetime and HttpPooledConnectionIdleTimeout to DocumentConventions

### PR DESCRIPTION
**Pull request by ppekrol**
### Issue link

https://issues.hibernatingrhinos.com/issue/RavenDB-19399

### Type of change

- New feature

### How risky is the change?

- Moderate 

### Backward compatibility

- Non breaking change

### Is it platform specific issue?

- No

### Documentation update

- This change requires a documentation update. Please mark the issue on YouTrack using `Documentation Required` tag.

### Testing 

- It has been verified by manual testing

### Is there any existing behavior change of other features due to this change?

- No

### UI work

- No UI work is needed

### CODE REVIEW

**Review by arekpalinski — APPROVED** [review]

File: src/Raven.Client/Documents/Conventions/DocumentConventions.cs [hunk 1]
```diff
@@ -42,7 +42,10 @@ public class DocumentConventions : Client.Conventions
         internal static readonly DocumentConventions DefaultForServer = new DocumentConventions
         {
             SendApplicationIdentifier = false,
-            MaxContextSizeToKeep = new Size(PlatformDetails.Is32Bits == false ? 8 : 2, SizeUnit.Megabytes)
+            MaxContextSizeToKeep = new Size(PlatformDetails.Is32Bits == false ? 8 : 2, SizeUnit.Megabytes),
+#if NETCOREAPP3_1_OR_GREATER
+            HttpPooledConnectionLifetime = TimeSpan.FromMinutes(3)
```
Comment (ayende) [inline src/Raven.Client/Documents/Conventions/DocumentConventions.cs:47]: Do we have to worry about performance issues here? If I understand properly, it means that every 3 minutes, we'll need to establish a new connection, no?

Comment (ppekrol) [reply src/Raven.Client/Documents/Conventions/DocumentConventions.cs:47]: This is for server-side and this is transparently handled by the framework. If connection is used, it is not being disconnected and reconnected. There might be some super small overhead, but server mostly gets TCP info via executor and switched directly to TCP, which is not affected.

Comment (ayende) [reply src/Raven.Client/Documents/Conventions/DocumentConventions.cs:47]: One place where this happens a _lot_ is the `forward to leader` scenario. We saw some cases where we have > 1K/sec requests in this scenario.

And the issue is that your connection lifetime is now 3 minutes. Assume that establishing a connection take a while (common in distributed scenarios), and now compare exchange or cluster wide transaction will suffer latency spike every 3 minutes, no?

Comment (ppekrol) [reply src/Raven.Client/Documents/Conventions/DocumentConventions.cs:47]: Yet, it is mechanism created by MS to protect for DNS and connectivity issues. I can turn it off, but then cluster is getting stuck permanently.

Comment (ayende) [reply src/Raven.Client/Documents/Conventions/DocumentConventions.cs:47]: I'm very happy to have this as a configuration option, my concern is the default setting.
Furthermore, don't we _already_ have handling for this scenario, with the renewal of the `http client` ?

**Pull request merged**